### PR TITLE
Add Github Action cache to speed up build docs workflow

### LINKED ISSUE #113: Use Github Action Caching to Speed Up Tests
More info: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #113

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -12,7 +12,15 @@ jobs:
...
         with:
           python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: "pyproject.toml"
+      - uses: actions/cache@v3
+        id: cache
+        with:
+          path: ~/.cache/pip
+          key: python-deps-${{ matrix.python-version }}-${{ hashFiles('pyproject.toml') }}-tests-v01
       - name: Install dependencies
```
Comment (gsheni) [inline .github/workflows/tests.yaml:22]:
```suggestion
      - name: Install dependencies (if cache has not changed)
```

**Comment by kmax12** [comment]
can you tell if it is actually speeding up the tests? it's a bit more complexity than I thought, so probably only worth it if the tests are speeding up significantly

**Comment by gsheni** [comment]
#### The Build Docs workflow did speed up.
- No Cache ~ 2m 38s (https://github.com/kmax12/gridstatus/actions/runs/3809303349/jobs/6480574975)
- Cache ~1m 56s (https://github.com/kmax12/gridstatus/actions/runs/3813177529/jobs/6486773457)

**Comment by gsheni** [comment]
I was unable to get the cache working for the unit tests. The cache will only be saved if the workflow succeed, but the unit tests are currently failing. 
I tried some tricks (disable fail-fast, silence pytest output) with no luck. **My guess is you would save ~23s per each unit test run per python version (~23 x 5 = 115)**

**Pull request closed without merging**